vnext/Microsoft.ReactNative/Fabric/Composition/SwitchComponentView.cpp

```diff
@@ -203,9 +203,25 @@ void SwitchComponentView::Draw() noexcept {
       thumbX = (trackMarginX + trackWidth - thumbRadius - thumbRadius - thumbMargin) * m_layoutMetrics.pointScaleFactor;
     }
 
-    m_thumbVisual.Size(
-        {thumbRadius * m_layoutMetrics.pointScaleFactor, thumbRadius * m_layoutMetrics.pointScaleFactor});
-    m_thumbVisual.Position({thumbX, thumbY});
+    // handles various mouse events
+    if (m_pressed && !switchProps->disabled) {
+      m_thumbVisual.AnimatePosition({thumbX - 0.8f, thumbY - 0.8f});
+    } else if (m_hovered && !switchProps->disabled) {
+      m_thumbVisual.Size(
+          {thumbRadius * m_layoutMetrics.pointScaleFactor + 0.8f,
+           thumbRadius * m_layoutMetrics.pointScaleFactor + 0.8f});
+      m_thumbVisual.Position({m_thumbVisual.Position().x - 0.8f, m_thumbVisual.Position().y - 0.8f});
+    } else if (m_pointerReleased && !switchProps->disabled) {
+      m_pointerReleased = false;
```

Comment:
You shouldn't change state in Draw.  Its possble that Draw could be called again, and you wouldnt want it to draw differently the 2nd time. 

Right now if would get called again on theme change (which isn't hooked up to anything yet).  In the future it'll also need to be called on graphic device lost too.

Reply:
ahh that makes sense, removed the state change from Draw and into onPointerEntered.